src/PowerShellEditorServices/Services/Symbols/SymbolsService.cs

```diff
@@ -284,7 +263,12 @@ static string[] GetIdentifiers(string symbolName, SymbolType symbolType, Command
                 return null;
             }
 
-            return AstOperations.FindReferencesOfSymbol(file.ScriptAst, foundSymbol).ToArray();
+            if (file.References.TryGetReferences(foundSymbol.SymbolName, out ConcurrentBag<SymbolReference> references))
+            {
+                return references;
+            }
```

Comment:
```suggestion
            if (file.References.TryGetReferences(foundSymbol.SymbolName, out ConcurrentBag<SymbolReference> references))
            {
return references.Where((r) =>
                    foundSymbol.SymbolType == r.SymbolType ||
                    (foundSymbol.SymbolType is SymbolType.Class or SymbolType.Enum or SymbolType.Type
                        && r.SymbolType is SymbolType.Class or SymbolType.Enum or SymbolType.Type)
                );
            }
```

Filter on symbol type. Will currently match ex. a similar function name and enum/class name.